### PR TITLE
feat: tag

### LINKED ISSUE #4794: Design system: Tags
Updated tags ready for review:

https://www.figma.com/file/2MLHeIeL6XPVi3Tc2DfFCr/%E2%9D%96-Leather-%E2%80%93-Design-System?type=design&node-id=13136-23521&mode=design&t=D87RVUYDDbUWp8ZU-4

Image

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8160562774), [Test report](https://leather-wallet.github.io/playwright-reports/feat/tags)

This PR adds the new Tag component with stories per the design spec (see issue).

*I also refactored the Callout to use a recipe with variants bc it seems easier to read imo, but can remove if people prefer data- params?

Screenshot 2024-03-04 at 7 54 56 PM

Screenshot 2024-03-04 at 7 55 09 PM

Screenshot 2024-03-04 at 7 58 58 PM

Screenshot 2024-03-04 at 8 23 42 PM

### CODE REVIEW

File: src/app/ui/components/callout/callout.tsx [hunk 1]
```diff
@@ -1,44 +1,51 @@
...
-  color: 'ink.text-primary',
-  textAlign: 'left',
-  width: '100%',
-
-  '&[data-variant="default"]': {
-    backgroundColor: 'ink.non-interactive',
-  },
-  '&[data-variant="error"]': {
-    backgroundColor: 'red.background-secondary',
+const calloutRecipe = cva({
```
Comment (pete-watters) [inline src/app/ui/components/callout/callout.tsx:9]: Nice work adding the recipe, I think it is good and worth doing where we can 👍

**Review by pete-watters — APPROVED** [review]
Nice work 👍

File: src/app/ui/components/tag/tag.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  defaultVariants: {
+    variant: 'default',
+  },
+});
+
+type TagVariants = RecipeVariantProps<typeof tagRecipe>;
+
+export interface TagProps extends BoxProps {
+  icon?: React.JSX.Element;
+  label: string;
```
Comment (kyranjamie) [inline src/app/ui/components/tag/tag.tsx:55]: Should this be `ReactNode` ?

**Review by kyranjamie — APPROVED** [review]

File: src/app/ui/components/tag/tag.tsx (hunk 2 above)
Comment (fbwoolf) [reply src/app/ui/components/tag/tag.tsx:55]: O yeah, thx, I was experimenting with something and meant to change back. 👍

**Pull request merged**